ordered_product/tests.py

```diff
@@ -0,0 +1,147 @@
+        user_name=client0,
+        supplier_product_id=supplier_product0,
+        quantity=3)
+
+
+class TestOrderedProductModel():
+    @pytest.mark.django_db()
+    def test_save_and_delete_ordered_product(self, delivery_location0, supplier_product0, ordered_product0):
+        delivery_location0.save()
+        supplier_product0.save()
+        ordered_product0.save_ordered_product()
+        assert ordered_product0 in OrderedProduct.objects.all()
+        ordered_product0.delete_ordered_product()
+        assert ordered_product0 not in OrderedProduct.objects.all()
+
+    @pytest.mark.django_db()
+    def test_filter_by_id(self, delivery_location0, supplier_product0, ordered_product0):
+        delivery_location0.save()
+        supplier_product0.save()
+        ordered_product0.save_ordered_product()
+        assert ordered_product0 not in list(OrderedProduct.filter_id(1000))
+        filt = OrderedProduct.filter_id(ordered_product0.id)
+        assert ordered_product0 in list(filt)
+
+    @pytest.mark.django_db()
+    def test_filter_by_user_name(self, delivery_location0, supplier_product0, ordered_product0):
+        delivery_location0.save()
+        supplier_product0.save()
+        ordered_product0.save_ordered_product()
+        assert ordered_product0 not in list(OrderedProduct.filter_user_name(""))
+        filt = OrderedProduct.filter_user_name(ordered_product0.user_name)
+        assert ordered_product0 in list(filt)
+
+    @pytest.mark.django_db()
+    def test_filter_by_supplier_product_id(self, delivery_location0, supplier_product0, ordered_product0):
+        delivery_location0.save()
+        supplier_product0.save()
+        ordered_product0.save_ordered_product()
+        assert ordered_product0 not in list(OrderedProduct.filter_supplier_product_id(-1))
+        filt = OrderedProduct.filter_supplier_product_id(ordered_product0.supplier_product_id)
+        assert ordered_product0 in list(filt)
+
+    @pytest.mark.django_db()
+    def test_filter_delivery_location_id(self, delivery_location0, supplier_product0, ordered_product0):
+        delivery_location0.save()
+        supplier_product0.save()
+        ordered_product0.save_ordered_product()
+        assert ordered_product0 not in list(OrderedProduct.filter_delivery_location_id(-1))
+        filt = OrderedProduct.filter_delivery_location_id(ordered_product0.delivery_location_id)
+        assert ordered_product0 in list(filt)
+
+    @pytest.mark.django_db()
+    def test_validators(self, delivery_location0, supplier_product0, ordered_product0):
+        try:
+            delivery_location0.save()
+            supplier_product0.save()
+            ordered_product0.save_ordered_product()
+            tmp_ordered_product = ordered_product0
+            tmp_ordered_product.quantity = 0
+            tmp_ordered_product.save_ordered_product()
+        except ValidationError:
+            assert tmp_ordered_product not in OrderedProduct.objects.all()
```

Comment:
Is this where we plan to fail? What's actually failing here?

Reply:
I'll fix it.